nomad/fsm.go

```diff
@@ -214,10 +262,29 @@ func (n *nomadFSM) applyDeregisterJob(buf []byte, index uint64) interface{} {
 		panic(fmt.Errorf("failed to decode request: %v", err))
 	}
 
+	job, err := n.state.JobByID(req.JobID)
+	if err != nil {
+		n.logger.Printf("[ERR] nomad.fsm: DeleteJob failed: %v", err)
+		return err
+	}
+
 	if err := n.state.DeleteJob(index, req.JobID); err != nil {
 		n.logger.Printf("[ERR] nomad.fsm: DeleteJob failed: %v", err)
 		return err
 	}
+
+	if job.IsPeriodic() {
```

Comment:
We should remove this guard, and always clear out from periodicDispatcher and the launch table, since we may have taken a periodic job and made it non-periodic